src/Evolution/DiscontinuousGalerkin/SlopeLimiters/Minmod.cpp

```diff
@@ -95,10 +95,35 @@ bool limit_one_tensor(
       const bool has_neighbors = (externals.find(dir) == externals.end());
       if (has_neighbors) {
         const auto& neighbor_ids = element.neighbors().at(dir).ids();
-        ASSERT(neighbor_ids.size() == 1,
-               "Minmod does not yet support h-refinment");
-        // We've checked for a single neighbor, so just read first element:
-        const auto dir_key = std::make_pair(dir, *(neighbor_ids.cbegin()));
+
+        // Average neighbor_size toward dim, side:
+        // This average is independent of the tensor component `u`, so could be
+        // precomputed outside the loop over tensors. Changing the code to
+        // precompute the average may or may not be an optimization.
+        const double neighbor_size =
+            [&dim, &dir, &neighbor_ids, &neighbor_sizes ]() noexcept {
+          double size_cumul = 0.0;
+          for (const auto& id : neighbor_ids) {
+            size_cumul +=
+                gsl::at(neighbor_sizes.at(std::make_pair(dir, id)), dim);
+          }
+          return size_cumul / neighbor_ids.size();
+        }();
+
+        // Average neighbor_tensor mean toward dim, side:
```

Comment:
average normal to face

Reply:
No: this function averages the tensor element-averages over the N neighboring elements in direction `dim`, side `side`; there is no sense in which this quantity is normal to anything. 

Would it help if I rephrased as something like "Average the neighbor_tensor means over the different neighbors on the opposite side of the face" ?

Reply:
Ah yes, true.  Yes, the rephrasing is better

Reply:
Okay, I'll rephrase this (and the other one to match)